Cql/Cql.CqlToElm/ModelProvider.cs

```diff
@@ -276,15 +282,18 @@ internal static Elm.TypeSpecifier GetTypeSpecifierForElement(this ClassInfoEleme
         /// </summary>
         /// <remarks>This function checks a subtyping relationship, not whether in- or explicit casts may exist between the types.
         /// See https://cql.hl7.org/03-developersguide.html#type-testing for more information.</remarks>
-        public static bool IsSubtypeOf(this Elm.TypeSpecifier subType, Elm.TypeSpecifier superType, IModelProvider provider)
+        internal static bool IsSubtypeOf(this Elm.TypeSpecifier subType, Elm.TypeSpecifier superType, IModelProvider provider)
         {
-            if (superType == subType || superType == SystemTypes.AnyType)
```

Comment:
Initially, my thinking that `IsSubType` from the model's perspective should only return `true` when there actually is an unbroken inheritance chain from the type back to the `Any` type declared in the `elm-modelinfo-r1.xml`.  Neither `List` nor `Interval` are actually declared in any model.  I was not sure if argument-to-operand binding during signature resolution was matching based on `List` and `Interval` using `Subtype` conversion precedence, or a different kind.

Bryn later clarified in Zulip that the above logic is correct: *every* type should be considered a subtype of Any, regardless of whether it's in the model or not.

The spec also says this:
`the term subtype applies to all derived types, as well as the type itself.`

I introduced a test called `TypesAreNotSubtypesOfThemselves` that I think violates the spec language [here](https://cql.hl7.org/03-developersguide.html#type-testing), saying: 

>Note that because of the identity relationship above, the term subtype applies to all derived types, as well as the type itself. In the discussions that follow, if a definition must explicitly refer to only derived types, the term proper subtype will be used.

I think I will rather rename this current implementation `IsProperSubtype`, and then implement `IsSubType` that will immediately also check if `subType == superType` as above.

All tests will pass with these changes.